pyglider/seaexplorer.py

```diff
@@ -366,7 +370,21 @@ def raw_to_timeseries(indir, outdir, deploymentyaml, kind='raw',
                     )[:-1, :]
                     val2 = sensor_sub_grouped.select(sensorname).to_numpy()[:, 0]
                     val = _interp_gli_to_pld(sensor_sub_grouped, sensor, val2, indctd)
-                val = val[indctd]
+                if interpolate and not np.isnan(val).all():
+                    time_original = sensor.select('time').to_numpy()[:, 0]
+                    time_var = time_original[np.where(~np.isnan(val))[0]]
+                    var_non_nan = val[np.where(~np.isnan(val))[0]]
+                    time_timebase = sensor.select('time').to_numpy()[indctd, 0]
+                    if val.dtype == "<M8[us]":
+                        # for datetime, must convert to numerical, interpolate, then convert back
+                        us_since_1970 = (var_non_nan - np.datetime64("1970-01-01")).astype(int)
+                        val_int = np.interp(time_timebase.astype(float), time_var.astype(float), us_since_1970)
+                        val_us = val_int.astype("timedelta64[us]")
+                        val = np.datetime64("1970-01-01") + val_us
+                    else:
+                        val = np.interp(time_timebase.astype(float), time_var.astype(float), var_non_nan)
+                else:
+                    val = val[indctd]
```

Comment:
This may betray my ignorance with regards to testing, but why wouldn't line 387 here be covered by the tests? One of your tests covers the raw data with `interpolate=False`, which should include line 387, shouldn't it?

Reply:
It certainly should do. Not sure why codecov doesn't pick that up